nunaliit2-js/src/main/js/nunaliit2/n2.inputChangeDetector.js

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright (c) 2017, Geomatics and Cartographic Research Centre, Carleton 
+University
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without 
+modification, are permitted provided that the following conditions are met:
+
+ - Redistributions of source code must retain the above copyright notice, 
+   this list of conditions and the following disclaimer.
+ - Redistributions in binary form must reproduce the above copyright notice,
+   this list of conditions and the following disclaimer in the documentation
+   and/or other materials provided with the distribution.
+ - Neither the name of the Geomatics and Cartographic Research Centre, 
+   Carleton University nor the names of its contributors may be used to 
+   endorse or promote products derived from this software without specific 
+   prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
+ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE 
+LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR 
+CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF 
+SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+POSSIBILITY OF SUCH DAMAGE.
+
+*/
+
+/* 
+ * ------------------------------------------------------------------------------
+ * Description: Simple atlas plug-in that detects if an input field value is 
+ * empty or not. When value is not empty an input_detected class is added to the 
+ * element.
+ * ------------------------------------------------------------------------------
+ */
+
+;(function($,$n2){
+"use strict";
+
+//--------------------------------------------------------------------------
+var InputChangeDetector = $n2.Class('InputChangeDetector', {
+	
+	initialize: function(opts_){
+
+		var opts = $n2.extend({
+			config: null
+			,options: null
+		},opts_);
+		
+		var _this = this;
+
+		this.startDetector();
+	},
+	
+	startDetector: function(){
+		
+		$('body').change(function(){
+			$(this).find('input').each(function(){
+				var value = $(this).val();
+
+				if( value !== '' && !$(this).hasClass('input_detected') ){
+					$(this).addClass('input_detected');
+	
+				} else if( value === '' && $(this).hasClass('input_detected') ){
+					$(this).removeClass('input_detected');
+				};
+			});
+		});
+	}
+});
+
+$n2.inputChildDetector = new InputChangeDetector();
```

Comment:
Sounds great! 

I'll convert it into a utility. 

I will also need to add some code to handle when a user has the utitlity disabled. Since the transition animations are dependent on this utility updating input elements, I will need to add a way for those transitions to not occur if it's disabled. 

Will let you know when it's ready for a second review. 

Thanks,